src/models/StyledComponent.js

```diff
@@ -48,7 +48,10 @@ export default (ComponentStyle: Function) => {
           const subscribe = this.context[CHANNEL]
           this.unsubscribe = subscribe(nextTheme => {
             // This will be called once immediately
-            const theme = this.props.theme || nextTheme
+            const { defaultProps } = this.constructor
+            const defaultTheme = defaultProps && defaultProps.theme
+            const isDefaultTheme = defaultTheme && this.props.theme === defaultTheme
```

Comment:
Couldn't this be also `const isDefaultTheme = this.props.theme === defaultTheme`?